src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemResources.cs

```diff
@@ -2032,9 +2033,103 @@ internal override bool IsUnset
 
         #endregion Properties
     }
-}
 
+    internal class DeferredResourceReferenceList
```

Comment:
add sealed

Reply:
Why? There is no virtual members and it's an internal type.

Reply:
@miloush
[See](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1852)

Reply:
> Sealing types can improve performance.

Not very insightful...

Reply:
Also this assembly has [InternalsVisibleToAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute) so according to your link this should not be flagged.

Reply:
The answer to my question is here: https://github.com/dotnet/runtime/issues/49944. I am not objecting the suggestion in this case, just want to know why we are doing it.

Reply:
@miloush 
Why make a class unsealed if it has no heirs? When reading the code, you may be tempted to look for heirs.